src/gui/properties/speedwidget.cpp

```diff
@@ -0,0 +1,238 @@
+#include "speedwidget.h"
+
+#include <QDateTime>
+#include <libtorrent/session_status.hpp>
+
+#include "propertieswidget.h"
+#include "core/bittorrent/session.h"
+#include "core/bittorrent/sessionstatus.h"
+#include "core/preferences.h"
+
+SpeedWidget::SpeedWidget(PropertiesWidget *parent)
+    : QWidget(parent)
+    , m_properties(parent)
+{
+    m_layout = new QVBoxLayout(this);
+    m_layout->setContentsMargins(0, 0, 0, 0);
+
+    m_hlayout = new QHBoxLayout();
+    m_hlayout->setContentsMargins(0, 0, 0, 0);
+
+    m_period_label = new QLabel("<b>" + tr("Period:") + "</b>");
+
+    m_period_combobox = new QComboBox();
+    m_period_combobox->addItem(tr("1 Minute"));
+    m_period_combobox->addItem(tr("5 Minutes"));
+    m_period_combobox->addItem(tr("30 Minutes"));
+    m_period_combobox->addItem(tr("6 Hours"));
+
+    connect(m_period_combobox, SIGNAL(currentIndexChanged(int)), this, SLOT(on_period_change(int)));
+
+    m_graphs_button = new QToolButton();
+    m_graphs_button->setText(tr("Select Graphs"));
+    m_graphs_button->setPopupMode(QToolButton::InstantPopup);
+
+    m_graphs_menu = new QMenu();
+    m_graphs_menu->addAction(tr("Total Upload"));
+    m_graphs_menu->addAction(tr("Total Download"));
+    m_graphs_menu->addAction(tr("Payload Upload"));
+    m_graphs_menu->addAction(tr("Payload Download"));
+    m_graphs_menu->addAction(tr("Overhead Upload"));
+    m_graphs_menu->addAction(tr("Overhead Download"));
+    m_graphs_menu->addAction(tr("DHT Upload"));
+    m_graphs_menu->addAction(tr("DHT Download"));
+    m_graphs_menu->addAction(tr("Tracker Upload"));
+    m_graphs_menu->addAction(tr("Tracker Download"));
+
+    graphs_menu_actions = m_graphs_menu->actions();
+    m_graphs_signal_mapper = new QSignalMapper();
+
+    for (int id = UP; id <= TRACKER_DOWN; ++id) {
```

Comment:
After you append NB_COLUMNS then you change this to `for (int id = UP; id < NB_COLUMNS; ++id)`.
If in the future another ID is added, it will not be needed to come here and change the code.